dual_net.py

```diff
@@ -723,10 +730,61 @@ def freeze_graph_tpu(model_path):
 
         tf.train.Saver().restore(sess, model_path)
 
-    # Freeze the graph.
-    model_def = tf.graph_util.convert_variables_to_constants(
+    out_graph = tf.graph_util.convert_variables_to_constants(
         sess, sess.graph.as_graph_def(), output_names)
-    with tempfile.NamedTemporaryFile(mode='wb', delete=False) as f:
-        f.write(model_def.SerializeToString())
-        tmp_path = f.name
-    tf.gfile.GFile.Rename(tmp_path, model_path + '.pb')
+
+    metadata = get_model_metadata({
+        'engine': 'tpu',
+    })
+
+    atomic_write_model(out_graph, metadata, model_path)
+
+
+def get_model_metadata(metadata):
```

Comment:
this is a surprising getter.  maybe call it "make_model_metadata", or just two-step it with update()?
(e.g. 
```
metadata = {'engine' : 'tpu'}
metadata.update(get_model_metadata())
```
)